### DISCUSSION TITLE
(Capstone DV) Mengubah frequency data dalam bar plot pada shiny

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#572] [Q&A] [answered]
Jadi saya sedang membuat bar plot yang menunjukkan ranking performa saham. Frekuensi data yang saya miliki pada saat ini adalah dalam bentuk daily. Saat ini, bar plot saya terlihat seperti ini:
![image](https://user-images.githubusercontent.com/122977451/221126816-8c108cf6-db52-4c23-9892-2b11c293ce05.png)

Bar plot tersebut saya buat dari data sebagai berikut (dari all_stocks, menjadi mean_dr) :
![image](https://user-images.githubusercontent.com/122977451/221127158-6d9b7975-5d29-4402-b007-6d992579c2d1.png)
![image](https://user-images.githubusercontent.com/122977451/221127184-04bea18d-085b-4083-9e36-7c1c3be75bfd.png)

Pertanyaan saya, bagaimana caranya supaya data pada plot tersebut dapat menunjukkan performa tidak hanya secara daily, namun juga secara weekly dan monthly? Saya tahu ada fungsi floor_date yang dapat mengubah bentuk data menjadi weekly dan montly, namun dari pengertian saya, sebuah kolom atau bahkan data frame baru harus dibentuk kalau misalnya menggunakan fungsi tersebut. Apakah ada cara supaya saya bisa menggunakan plot yang saya miliki pada saat ini untuk menunjukkan plot weekly dan monthly di shiny?

Sekedar informasi tambahan (jika ini penting), pemilihan daily, weekly, dan monthly pada shiny akan dilakukan dengan selectInput.
EDIT: Tambahan lagi, berikut kode yang saya gunakan di ui dan juga server:
![image](https://user-images.githubusercontent.com/122977451/221152125-45d309ee-9479-475f-8060-aa92b6d10b1d.png)
![image](https://user-images.githubusercontent.com/122977451/221152165-7855e773-af44-4043-b9d0-633db2ff6a43.png)

Saya sedang mengetes apakah bisa menggunakan if statement pada bagian server, namun pada saat ini sedang ada masalah: Jika memilih Daily, plot tidak muncul, sedangkan muncul pesan error "Unsupported date-time class 'grouped_df'Unsupported date-time class 'tbl_df'Unsupported date-time class 'tbl'Unsupported date-time class 'data.frame' " jika memilih Weekly.

Mohon bantuannya, terima kasih

EDIT 2: Pada saat ini saya memutuskan untuk membuat 3 plot yang berbeda untuk menunjukkan frekuensi data yang saya inginkan menggunakan floor_date. Namun bagi yang memiliki solusi berbeda untuk masalah saya, boleh di share ya kalau berkenan hehe

**Comment by idoali** [comment, marked as answer]
Terima kasih pak atas pertanyaannya.

Intinya kita perlu membuat kolom baru yang berisikan nomor pekan (1-52) dan nama bulan pak. Kolom nomor pekan dan bulan ini akan kita pakai untuk `group_by` nantinya pak. 

Jadi step-by-step nya sendiri yang kami pikirkan kurang lebih seperti ini pak : 
- Buat kolom baru `week` dan `month` menggunakan fungsi `week()` dan `month()` dari library `lubridate`
- Buat input baru di `ui.R` yang pilihannya adalah daily, weekly atau monthly (ini sudah bapak lakukan dengan `input$Frame`). Input ini akan kita simpan sebagai `input$Frame`, katakanlah
- Lakukan `group_by` berdasarkan `input$Frame` di file `server.R`. Kita summarise dan simpan di dalam objek baru, yang namanya katakanlah `data_freq`
- `data_freq` inilah yang akan kita masukkan ke dalam `ggplot()`

Catatan : Agar lebih mudah, saat membuat plot data yang kita masukkan adalah data yang sudah siap untuk divisualisasikan. Jadi tidak ada lagi proses pengambilan nama bulan atau pekan saat membuat `ggplot`.
